### DISCUSSION TITLE
HP Z230 Desktop - L51 1.63

### DISCUSSION THREAD

**Discussion by Giinkie** [xCuri0/ReBarUEFI#131] [General]
Hello guys, I have the HP Z230 PC and tried so many times to flash my bios, but it doesn't work, or i a little bit stupid)

I check this amazing repository and now need you're help to make it work.

1. - i successfully add NVME driver for this PC with this method(with GRUB) - https://winraid.level1techs.com/t/success-hp-z230-tower-workstation/38156
2. - any other methods to flash my bios without physical programming - didn't work. Not FPTW, not Afudos, not HPQFlash(HP use it to update).
3. Maybe if we reflash only the section of region in BIOS, like in method above(NVME), that will work?
4. And the last - I don't have control of 4G Decoding in BIOS, but know than it will activate(check it on AMIBCP, and successfully installation and working Hackintosh on iGPU, who need it to work properly)

Thanks for reading this, hope someone help me)
P.S Sorry from my bad English, really

**Comment by xCuri0** [comment]
@Giinkie So try the same method they used for Nvme DXE ? All you have to is use rebar ffs and set the length to be that of ReBar ffs instead which is even smaller than Nvme btw.


Obviously there's a high risk of brick if you do this wrong considering you don't have a flasher. But no reason it won't work if Nvme does
